scanpy/preprocessing/_pca.py

```diff
@@ -189,12 +238,12 @@ def pca(
     elif issparse(X) and zero_center:
         from sklearn.decomposition import PCA
 
-        if svd_solver == "auto":
-            svd_solver = "arpack"
-        if svd_solver not in {'lobpcg', 'arpack'}:
+        if svd_solver is None:
+            svd_solver = 'arpack'
```

Comment:
that’ll mean that `changed` will be set to true later, resulting in a warning when using the default.

as said, please make sure that leaving the default doesn’t result in a warning.

you’re almost there, you just need to leave this as `None` until the point where you can actually determine the final solver